src/bin/ion/commands/beta/generate/utils.rs

```diff
@@ -20,49 +20,144 @@ pub struct Import {
     pub(crate) type_name: String,
 }
 
-/// Represent the programming language for code generation.
-#[derive(Debug, Clone, PartialEq)]
-pub enum Language {
-    Rust,
-    Java,
+pub trait Language {
```

Comment:
(🗺️ PR tour) Adds a trait implementation of `Language` instead of an enum (#71). This trait will be implemented for different programming languages and can be used as a generic on `CodeGenerator` to trigger specific behavior based on programming language.